wagtail/test/utils/template_tests.py

```diff
@@ -0,0 +1,77 @@
+from typing import Dict, List, Union
+
+from django.test import SimpleTestCase
+
+from .wagtail_tests import WagtailTestUtils
+
+
+class AdminTemplateTestUtils:
+    base_breadcrumb_items = [{"label": "Home", "url": "/admin/"}]
+
+    def assertBreadcrumbsItemsRendered(
+        self: Union[WagtailTestUtils, SimpleTestCase],
```

Comment:
Hmm, annoying that this doesn’t convey the need for the `WagtailTestUtils` instance to also inherit from `TestCase`. I wonder if there’s a better way to annotate this.

Reply:
Yeah unfortunately I couldn't find a better way to do it. The `WagtailTestUtils` class is basically a mixin, and it doesn't assume which `TestCase` base class you're going to use it with. For example, you can use it with a `SimpleTestCase` or the more featured `TestCase` depending on your needs, but we can't enforce this unless we make `WagtailTestUtils` extend from that (at which point it might break existing tests because they now inherit the `TestCase` twice).

I just opted for the most basic class needed (`SimpleTestCase`) and used `Union` with `WagtailTestUtils` to at least get VSCode to understand where the methods come from.

Alternatively you could define a class just for the sake of typing, e.g.

```py
class WagtailTestUtilsTestCase(WagtailTestUtils, SimpleTestCase):
    ...
```

and use it to annotate `self`, but I don't really like doing this because it would actually define the class (unlike a TypeScript interface that doesn't produce runtime code). VSCode's type shims tend to use this approach with `.pyi` files.